group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -20635,12 +21067,10 @@ barTopicName, new TopicMetadata(barTopicId, barTopicName, 3)
                 ),
                 // Remove regex.
                 List.of(GroupCoordinatorRecordHelpers.newConsumerGroupRegularExpressionTombstone(groupId, "foo*")),
-                // Updated subscription metadata.
-                List.of(GroupCoordinatorRecordHelpers.newConsumerGroupSubscriptionMetadataRecord(groupId, Map.of(
-                    barTopicName, new TopicMetadata(barTopicId, barTopicName, 3)
-                ))),
                 // Bumped epoch.
-                List.of(GroupCoordinatorRecordHelpers.newConsumerGroupEpochRecord(groupId, 11, 0))
+                List.of(GroupCoordinatorRecordHelpers.newConsumerGroupEpochRecord(groupId, 11, computeGroupHash(Map.of(
+                    barTopicName, barTopicHash
+                ))))
             ),
             result.records()
```

Comment:
Not related to this line but it would be great if we could add more tests for the cache cleaning (e.g. when a topic is updated or deleted).